action/protocol/staking/handlers.go

```diff
@@ -691,20 +691,21 @@ func (p *Protocol) handleCandidateRegister(ctx context.Context, act *action.Cand
 	// cannot collide with existing identifier
 	candID := owner
 	if !featureCtx.CandidateIdentifiedByOwner {
-		id, err := p.generateCandidateID(owner, blkCtx.BlockHeight, csm)
-		if err != nil {
+		// cannot collide with existing identifier
+		if csm.GetByIdentifier(owner) != nil {
 			return log, nil, &handleError{
-				err:           errors.Wrap(err, "failed to generate candidate ID"),
+				err:           ErrInvalidOwner,
 				failureStatus: iotextypes.ReceiptStatus_ErrCandidateAlreadyExist,
 			}
 		}
-		candID = id
-		if csm.GetByIdentifier(candID) != nil {
```

Comment:
The reason for deletion here is that there is no need to check for ID conflicts anymore, as this is already ensured in generateCandidateID.